### DISCUSSION TITLE
A conceptually correct way to set up Asset materialized on schedule

### DISCUSSION THREAD

**Discussion by DanilaBaklazhenkoWolt** [dagster-io/dagster#17597] [Q&A] [answered]
Hello! Seeking conceptual support here on how to correctly set up software-defined assets and materialize them on some schedule.
I've approached it by defining the asset, creating a user-defined job for materializing it, and creating a schedule for this job.

However, I'm facing some issues:

- Clicking the materialize button on the asset in the UI triggers a generic materialization job, not the user-defined one. To start the user-defined job, I have to manually find and trigger it in the UI.
- The generic job lacks the tags of the user-defined job, impacting my Slack notifications on failure which is supposed to use tags as metadata source.
- More crucially, the user-defined job neither inherits nor lets me set a custom backfill policy defined on the asset, making it unusable for custom backfill policies.

Is it supposed to be like that by design or am I doing something wrong?

**Comment by sryza** [comment, marked as answer]
> Clicking the materialize button on the asset in the UI triggers a generic materialization job, not the user-defined one. To start the user-defined job, I have to manually find and trigger it in the UI.

This is the expected behavior, although I see how it could be confusing in your case.  One potential solution here might be for us to allow you to specify that certain assets are only materializable through their job.

Separately, we're working on adding a cron schedule-based auto-materialize policy, which would allow scheduling these assets without using jobs. Would that make sense in your situation?

> The generic job lacks the tags of the user-defined job, impacting my Slack notifications on failure which is supposed to use tags as metadata source.

If there were a way to set these tags on individual assets, would that make sense to you? Or do they apply more to the job than to the asses?

> More crucially, the user-defined job neither inherits nor lets me set a custom backfill policy defined on the asset, making it unusable for custom backfill policies.

We are tracking this issue here: https://github.com/dagster-io/dagster/issues/17404

**Reply by DanilaBaklazhenkoWolt** [reply]
Hey, thank you for the quick response! 

> One potential solution here might be for us to allow you to specify that certain assets are only materializable through their job.

> Separately, we're working on adding a cron schedule-based auto-materialize policy, which would allow scheduling these assets without using jobs. Would that make sense in your situation?

Both options will do the trick for my case. Small feedback from my side is that I expected the asset should be either tied to a job or completely decoupled. Now it seems like it is stuck somewhere in between and that brings some confusion.

> If there were a way to set these tags on individual assets, would that make sense to you? Or do they apply more to the job than to the asses?

Yeah, that definitely makes sense. One more thing that I tried to do is to put those tags as metadata on an asset and push it to a job, but I couldn't find an easy way to do it without fetching the asset definition from `Definitions`.
